diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/StringCaseUtils.kt

```diff
@@ -46,6 +46,7 @@ fun String.isLowerSnakeCase(): Boolean = this.matches("(([a-z]+)_*)+[a-z0-9]*".t
  *
  * @return converted string
  */
+@Suppress("STRING_CONCATENATION")
```

Comment:
Well, fix the code according to the new rule :)

Reply:
There's a method chain that returns a string. I would not want to fix it. It will be ugly.
return this[idx].toUpperCase().toString() -> return "${this[idx].toUpperCase().toString()}"

Reply:
But there is the expression `this[idx].toUpperCase().toString() + convertUnknownCaseToUpperSnake(this.substring(idx + 1))`, which would be converted to `"${this[idx].toUpperCase()}${convertUnknownCaseToUpperSnake(this.substring(idx + 1))}"`. My concern is, that if we start suppressing the rule right away, there may be something in the rule that needs tuning. @akuleshov7 could you suggest whether this is desired behaviour?

Reply:
Let's change the Rule: if the `r-value` is not a constant string then DO NOT change them to template.
BUT if the the r-value is a constant string, it should be concatenated: `a.toString()` + " kjfgjfhgkjhdfg" -> "$a kjfgjfhgkjhdfg".

Please also reflect it in the codestyle!